gsa/src/web/entity/icon/editicon.js

```diff
@@ -25,18 +25,29 @@ import {getEntityType, typeName} from 'gmp/utils/entitytype';
 
 import PropTypes from 'web/utils/proptypes';
 import withCapabilities from 'web/utils/withCapabilities';
-
+import Capabilities from 'gmp/capabilities/capabilities';
+import {useGetCaps} from 'web/pages/tasks/graphql';
 import EditIcon from 'web/components/icon/editicon';
 
 const EntityEditIcon = ({
-  capabilities,
   displayName,
   entity,
   name,
   title,
   onClick,
   ...props
 }) => {
+  let capabilities;
+
+  const query = useGetCaps();
+  const {data} = query();
+
+  if (isDefined(data)) {
+    capabilities = new Capabilities(data.capabilities);
+  } else {
+    capabilities = props.capabilities;
+  }
+
```

Comment:
If you copy & paste something more then two times it really needs to be abstracted in some kind. Not sure if a function can be used here or it needs to be a react hook (which is actually also a function just the name must start with use).